### DISCUSSION TITLE
Proposed Feature: Detailed Scoring Feedback

### DISCUSSION THREAD

**Discussion by JoshuaKaplan1** [marod424/tamarisk#13] [Ideas]
I believe we should create a feature that provides detailed instructions and information regarding errors as to why students did not receive all points. It should give students what part of the code doesn't work, and perhaps provide ideas as to how to fix it (i.e. use a while loop instead of a for loop, input a semicolon, missing a curly bracket, etc). This feature could also allow students to go to each issue and see suggestions, feedback, and what error they would receive. The feature could allow the teacher to grade based on how many errors the student makes, as opposed to the student automatically receiving a 100 when they eventually figure out the code.

**Comment by marod424** [comment]
This is a great idea, thanks for the proposal.
